### PR TITLE
Rough update to KHR_lights support to match current spec

### PR DESCRIPTION
**Pull request by MiiBond**
Updated to match current spec.
KhronosGroup/glTF#1223

### CODE REVIEW

**Comment by donmccurdy** [comment]
For the renderer light settings, I think it's ok just to add:

```js
// The KHR_lights extension should use physically-correct lighting mode.
renderer.physicallyCorrectLights = true;
```

... to the entire example, no need to turn it on/off when the extension is selected.

**Comment by mrdoob** [comment]
@donmccurdy looks good?

**Comment by donmccurdy** [comment]
@mrdoob we should plan this one for r92, more likely, it will about a week before we've answered a couple last questions about this extension (partly my fault on that 😅).

**Comment by donmccurdy** [comment]
@MiiBond I think i've come around for having ambient lights on the scene...  do you want to include that change in this PR?

**Comment by MiiBond** [comment]
@donmccurdy I think this pull request handles the ambient light on the scene, doesn't it? Or did you mean something else?
This pull request doesn't attempt to limit the number of ambient lights or prevent them from being on a node or anything but I gather we're not relying too much on the loader validating the glTF.

**Comment by MiiBond** [comment]
It did just occur to me that the handling of direction for `SpotLight` and `DirectionalLight` are wrong. In three.js, the direction doesn't come from the matrix, right? It's dependent on the position and target of the light?

**Comment by donmccurdy** [comment]
> I think this pull request handles the ambient light on the scene, doesn't it? Or did you mean something else?

Oh! My mistake, I'd expected to see something in your PR checking for a light in the scene, but I guess that part was already here. 

> In three.js, the direction doesn't come from the matrix, right? It's dependent on the position and target of the light?

Hmm yeah. If we want a spotlight to point correctly in a direction relative to its local transform, even if the node is animated, perhaps we do this:

```js
light.target.position.set( 0, 0, 1 );
light.add( light.target );
```

**Comment by MiiBond** [comment]
@donmccurdy Yeah, I think that works.
I've updated the pull request.

**Comment by mrdoob** [comment]
Can we replace the current `Monster.gltf` model with the one with a light instead of having 2 `Monster.gltf`?

**Comment by donmccurdy** [comment]
We could drop `glTF-Embedded/Monster/*`, to just keep the current count.

For a future cleanup PR, I don't think we need to have all of `glTF`, `glTF-Embedded`, and `glTF-Binary` for every model — has been a long time since we've seen any bugs related to container structure. Keeping extensions around that add functionality like pbrSpecularGlossiness and lights seems more important.

File: examples/webgl_loader_gltf_extensions.html [hunk 1]
```diff
@@ -202,14 +202,13 @@
 					var ambient = new THREE.AmbientLight( 0x222222 );
 					scene.add( ambient );
 
-					var directionalLight = new THREE.DirectionalLight( 0xdddddd );
+					var directionalLight = new THREE.DirectionalLight( 0xdddddd, 4 );
```
Comment (donmccurdy) [inline examples/webgl_loader_gltf_extensions.html]: Why the intensity increase here?

**Review by donmccurdy — APPROVED** [review]
Looks good — We can remove some other example models in another PR.

File: examples/webgl_loader_gltf_extensions.html (hunk 1 above)
Comment (MiiBond) [reply examples/webgl_loader_gltf_extensions.html]: I think so, yeah. That was a while ago...

File: examples/js/loaders/GLTFLoader.js [hunk 2]
```diff
@@ -261,7 +262,14 @@ THREE.GLTFLoader = ( function () {
 
 				case 'spot':
 					lightNode = new THREE.SpotLight( color );
-					lightNode.position.set( 0, 0, 1 );
+					// Handle spotlight properties.
+					var spot = light.spot || {};
```
Comment (mrdoob) [inline examples/js/loaders/GLTFLoader.js]: How about doing this instead?

```js
var spot = light.spot || { innerConeAngle: 0, outerConeAngle: Math.PI / 4.0 };
lightNode.angle = spot.outerConeAngle;
lightNode.penumbra = 1.0 - ( spot.innerConeAngle / spot.outerConeAngle );
lightNode.target.position.set( 0, 0, 1 );
lightNode.add( lightNode.target );
```

File: examples/js/loaders/GLTFLoader.js [hunk 3]
```diff
@@ -261,7 +262,14 @@ THREE.GLTFLoader = ( function () {
...
+					var innerConeAngle = spot.innerConeAngle !== undefined ? spot.innerConeAngle : 0;
+					var outerConeAngle = spot.outerConeAngle !== undefined ? spot.outerConeAngle : Math.PI / 4.0;
+					lightNode.angle = outerConeAngle;
+					lightNode.penumbra = 1.0 - innerConeAngle / outerConeAngle;
+					lightNode.target.position.set( 0, 0, 1 );
```
Comment (mrdoob) [inline examples/js/loaders/GLTFLoader.js]: The target of a spot is `0, 0, 1` by default?

Comment (donmccurdy) [reply examples/js/loaders/GLTFLoader.js]: The local direction of the glTF light is always +Z; the only way to rotate it is by rotating the object or its parents in the scene graph.

File: examples/js/loaders/GLTFLoader.js (hunk 2 above)
Comment (donmccurdy) [reply examples/js/loaders/GLTFLoader.js]: `innerConeAngle` and `outerConeAngle` are not required to be present, so their default values must be set independently. @MiiBond is the `spot` property required? Perhaps:

```
light.spot = light.spot || {};
light.spot.innerConeAngle = light.spot.innerConeAngle !== undefined ? light.spot.innerConeAngle : 0;
light.spot.outerConeAngle = light.spot.innerConeAngle !== undefined ? light.spot.innerConeAngle : Math.PI / 4.0;
// ...
```

Comment (MiiBond) [reply examples/js/loaders/GLTFLoader.js]: `spot` isn't required, no.

I'm not sure I follow what you're suggesting with your change. Are you just suggesting to get rid of the variable declarations within the case statement?

Comment (donmccurdy) [reply examples/js/loaders/GLTFLoader.js]: I think that is the idea; i'm happy with or without that change.

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**